### PR TITLE
feat(query): jsonb parser optimize

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. optimize JSON parser
2. add JSON parse test cases
3. support `f64::NAN`, `f64::INFINITY`, `f64::NEG_INFINITY`

part of #7668

### CODE REVIEW

File: src/common/jsonb/src/number.rs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+}
+
+impl PartialEq<Number> for &Number {
+    #[inline]
+    fn eq(&self, other: &Number) -> bool {
+        (*self).eq(other)
+    }
+}
+
+impl Eq for Number {}
```
Comment (TCeason) [inline src/common/jsonb/src/number.rs:192]: I’m not sure why the impl is empty? Other codes lgtm.

Comment (b41sh) [reply src/common/jsonb/src/number.rs:192]: as `Eq` inheritance `PartialEq`, leave it empty is ok.

**Review by TCeason — APPROVED** [review]
LGTM

File: src/common/jsonb/src/number.rs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+            (_, _) => {
+                let l = OrderedFloat(self.as_f64().unwrap());
+                let r = OrderedFloat(other.as_f64().unwrap());
+                l.cmp(&r)
+            }
+        }
+    }
+}
+
+impl Debug for Number {
```
Comment (andylokandy) [inline src/common/jsonb/src/number.rs:244]: Why not to use derive?

**Review by andylokandy — APPROVED** [review]

File: src/common/jsonb/src/number.rs (hunk 2 above)
Comment (b41sh) [reply src/common/jsonb/src/number.rs]: modified

**Pull request merged**